porch/pkg/repository/repository.go

```diff
@@ -59,6 +59,10 @@ type PackageRevision interface {
 
 	// GetUpstreamLock returns the kpt lock information.
 	GetUpstreamLock() (kptfile.Upstream, kptfile.UpstreamLock, error)
+
+	// GetLock returns the current revision's lock information.
+	// This will be the upstream info for downstream revisions.
```

Comment:
I think this is a little bit misleading. As we discussed in our previous offline chat, `parent` only points to the upstream info if the current revision is a fresh clone. If the current revision was produced by editing/copying, then it points to the revision it was copied from (which can be different from upstream). 

Could we capture that distinction, either in this comment or in a comment above the git package implementation of this method?

Reply:
Discussed offline, renaming `parent` sounds like a great idea :)

Reply:
Cleaned it up and with some renaming the new code needs very less explanation. PTAL.